File: repo/query.go
package repo

import "github.com/Beriholic/wd/repo/model"

func MockQuery() (*model.Stardict, error) {
	return QueryWord("rust")
}

func QueryWord(wordName string) (*model.Stardict, error) {
	word := &model.Stardict{}

	err := DB.Model(&model.Stardict{}).Where("word = ?", wordName).First(&word).Error
	if err != nil {
		return nil, err
	}

	return word, nil
}

File: src/repo/query.rs
use crate::repo::get_db_connection;
use crate::repo::model::{Stardict, Word};
use color_eyre::Result;
use sqlx::query_as;

pub async fn query_word(name: &str) -> Result<Option<Word>> {
    let db = get_db_connection().await?;

    let db_word: Stardict = query_as(
        "SELECT * FROM stardict WHERE word == $1",
    ).bind(name).fetch_one(&db).await?;

    db.close().await;
    Ok(Some(db_word.into()))
}

#[cfg(test)]
mod tests {
    use crate::repo::query::query_word;

    #[tokio::test]
    async fn test_query_word() {
        let result = query_word("rust").await;
        let word = result.unwrap_or(None);

        assert_eq!(word.is_some(), true);
        assert_eq!(word.unwrap().name, "rust");
    }
}

